### DISCUSSION TITLE
Nesting Windows

### DISCUSSION THREAD

**Discussion by whereswalden** [glzr-io/glazewm#66] [Q&A] [answered]
In your example gif, at one point you have two windows, horizontal layout 50/50, then the rightmost window spawns a child window beneath it, Then in the bottom rightmost window, it splits horizontally. H 50%, V 25% > H 12.5%, H 12.5%.
In what manner of shortcuts did you achieve this organization? Is this specific to the app/program you are focused upon?
![image](https://user-images.githubusercontent.com/50717878/166260463-3a2d7e2f-6313-43c9-b535-bca8bdc9d629.png)

**Comment by w5922xd** [comment, marked as answer]
@whereswalden I had a similar question but think I have a grasp on it now. 

- Start with one application open. 
- Hit alt+shift+v to switch to Horizontal layout mode. 
- Open another application (Note: not a browser tab, it has to be a separate process). It should open next to it on the right. 
- Hit alt+v to switch to Vertical layout mode. 
- Open another application (Note: not a browser tab, it has to be a separate process). It should open below the second window. 
- And so on. 

What really threw me off was I was trying all this out with browser tabs. The behavior only impacts opening a new process and doesn't apply to existing ones.

**Reply by cloudhan** [reply]
This works for newly opened window, any way to apply to existing windows?

**Comment by maphew** [comment]
> The behavior only impacts opening a new process and doesn't apply to existing ones.

Ah, so that's why Windows Explorer windows don't follow the vertical/horizontal mode shift. Speaking of which: how to tell what mode is currently active?

**Reply by Purple-Tentacle** [reply]
Add this component to your config
`- type: "tiling direction"
      label_horizontal: "⮂"
      label_vertical: "⮁"
      background: "#ffffff33"
      margin: "0 4px"
      padding: "0 8px"`

**Comment by tibicen** [comment]
Is there any way of automating this behaviour based on width and height of the current window? "Automatic splitting" Bsd style?

**Reply by SpeedaRJ** [reply]
```python
import asyncio
import websockets
import json



async def get_status(web_socket):
    json_response = json.loads(await web_socket.recv())
    print(json_response)
    return json_response["success"]



async def main():
    uri = "ws://localhost:6123"

    async with websockets.connect(uri) as websocket:
        await websocket.send("sub -e focus_changed")

        while True:
            response = await websocket.recv()
            json_response = json.loads(response)

            if not json_response["data"]:
                continue

            try:
                containerData = json_response["data"]["focusedContainer"]
                w, h = containerData["width"], containerData["height"]

                await websocket.send("query tiling-direction")
                tilingDirection = json.loads(await websocket.recv())["data"]["tilingDirection"]

                if w < h:
                    await websocket.send('command set-tiling-direction vertical')
                    if not await get_status(websocket):
                        await websocket.send('command set-tiling direction vertical')
                        if not await get_status(websocket) and tilingDirection == "horizontal":
                            await websocket.send('command toggle-tiling-direction')
                            get_status(websocket)
                elif w > h:
                    await websocket.send('command set-tiling-direction horizontal')
                    if not await get_status(websocket):
                        await websocket.send('command set-tiling direction horizontal')
                        if not await get_status(websocket) and tilingDirection == "vertical":
                            await websocket.send('command toggle-tiling-direction')
                            get_status(websocket)
            except Exception as e:
                print(f"Error: {e}")


if __name__ == "__main__":
    asyncio.run(main())
```

I know this is a late reply (I just recently started using GlazeWM), I use the following script, that is more or less a combination of the two linked in the README of the project.
I then run it by adding the following do the `config.yaml` to the `startup_commands` list (with absolute paths):
```yaml
'shell-exec C:\<PYTHON_PATH>\pythonw.exe C:\%USERPROFILE%\.glzr\glazewm\autotile.py'
```

**Comment by teepoat** [comment]
Let's say I have 3 windows opened in horizontal layout mode. Can I create a workspace where first window is 50% horizontal 100% vertical size, and other two are in the right each 50% vertical size? Or the only way is to add window in vertical layout mode?
